javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -100,38 +100,45 @@ public static String annotationSimpleName(AnnotationMirror annotation) {
      */
     public static boolean areSame(
             /*@Nullable*/ AnnotationMirror a1, /*@Nullable*/ AnnotationMirror a2) {
-        if (a1 != null && a2 != null) {
-            if (a1 == a2) {
-                return true;
-            }
-            if (annotationName(a1) != annotationName(a2)) {
-                return false;
-            }
-
-            Map<? extends ExecutableElement, ? extends AnnotationValue> elval1 =
-                    getElementValuesWithDefaults(a1);
-            Map<? extends ExecutableElement, ? extends AnnotationValue> elval2 =
-                    getElementValuesWithDefaults(a2);
+        if (a1 == a2) {
+            return true;
+        }
+        if (a1 == null || a2 == null) {
+            return false;
+        }
 
-            return elval1.toString().equals(elval2.toString());
+        if (!areSameIgnoringValues(a1, a2)) {
```

Comment:
I would probably keep the previous `annotationName(a1) != annotationName(a2)`. The method call doesn't add much readability and adds redundant checks. But I'm fine with changing it, if you feel this makes it much easier to read.